staging/src/k8s.io/client-go/examples/leader-election/main.go

```diff
@@ -37,40 +37,58 @@ import (
 	"k8s.io/klog"
 )
 
-// main demonstrates a leader elected process that will step down if interrupted.
+func buildConfig(kubeconfig string) (*rest.Config, error) {
+	if kubeconfig != "" {
+		cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
 func main() {
 	klog.InitFlags(nil)
+
+	var kubeconfig string
+	var leaseLockName string
+	var leaseLockNamespace string
+	var id string
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	// we can use pod name as holder identity
+	flag.StringVar(&id, "id", os.Getenv("POD_NAME"), "the holder identity name")
```

Comment:
The README doesn't show how to run the example in a pod. I wouldn't leave the default to be empty.